Transaction statements not visible in pg_stat_statements view

kritika-0601 opened the pull request:
Currently Babelfish does not show transaction queries inside the pg_stat_statements table. After my change now it will be visible in the view.
pg_stat_statements displays planning and execution statistics for all SQL statements executed by a server. Some of the statements are not being tracked by it. Mainly because subsequent processutility hooks after bbf_processutilty are not being called, what happens is inside bbf_processUtility, babelfish specific function is called and it does not trigger the subsequent hooks due to which query statistics are not being stored in the table. 
Created a hook for that and called it inside relevant ProcessUtility function.

### Description

[Describe what this change achieves]
 
### Issues Resolved

[List any issues this PR will resolve]
 
### Check List

- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the PostgreSQL license, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

Commit:
Transaction statements not visible in pg_stat_statements view

Subsequent processutility hooks after bbf_processutilty are not being called.
Created a hook for that and called it inside relevant
ProcessUtility function.

Signed-off-by: Kritika <kritikaq@amazon.com>

kritika-0601 committed:
Merge branch 'babelfish-for-postgresql:BABEL_3_X_DEV__PG_15_X' into pg_stat_statements-transactions

surendravishnoi commented on src/backend/tcop/utility.c:
```diff
@@ -1656,8 +1665,10 @@ ProcessUtilitySlow(ParseState *pstate,
 			case T_CreateFunctionStmt:	/* CREATE FUNCTION */
-				address = CreateFunction(pstate, (CreateFunctionStmt *) parsetree);
-				break;
+				{
```
why this change?

surendravishnoi commented on src/backend/tcop/utility.c:
```diff
@@ -601,91 +602,99 @@ standard_ProcessUtility(PlannedStmt *pstmt,
-								SetQueryCompletion(qc, CMDTAG_ROLLBACK, 0);
-						}
-						break;
+						case TRANS_STMT_PREPARE:
```
please make sure we only have diff for actual changes, indentation is creating unnecessary diff

surendravishnoi commented on src/backend/tcop/utility.c:
```diff
@@ -601,91 +602,99 @@ standard_ProcessUtility(PlannedStmt *pstmt,
-				TransactionStmt *stmt = (TransactionStmt *) parsetree;
-
-				switch (stmt->kind)
+				if(NestedTranCount > 0 || (sql_dialect == SQL_DIALECT_TSQL && !IsTransactionBlockActive()))
```
this should not be here, only inside hook

Commit:
Transaction statements not visible in pg_stat_statements view

Subsequent processutility hooks after bbf_processutilty are not being called.
Created a hook for that and called it inside relevant
ProcessUtility function.

Signed-off-by: Kritika <kritikaq@amazon.com>

Commit:
Transaction statements not visible in pg_stat_statements view

Subsequent processutility hooks after bbf_processutilty are not being called.
Created a hook for that and called it inside relevant
ProcessUtility function.

Signed-off-by: Kritika <kritikaq@amazon.com>

Commit:
retrigger tests

Commit:
Transaction statements not visible in pg_stat_statements view

Subsequent processutility hooks after bbf_processutilty are not being called.
Created a hook for that and called it inside relevant
ProcessUtility function.

Signed-off-by: Kritika <kritikaq@amazon.com>

Commit:
Transaction statements not visible in pg_stat_statements view

Subsequent processutility hooks after bbf_processutilty are not being called.
Created a hook for that and called it inside relevant
ProcessUtility function.

Signed-off-by: Kritika <kritikaq@amazon.com>

kritika-0601 committed:
Merge branch 'babelfish-for-postgresql:BABEL_3_X_DEV__PG_15_X' into pg_stat_statements-transactions

Commit:
Transaction statements not visible in pg_stat_statements view

Subsequent processutility hooks after bbf_processutilty are not being called.
Created a hook for that and called it inside relevant
ProcessUtility function.

Signed-off-by: Kritika <kritikaq@amazon.com>

Commit:
Transaction statements not visible in pg_stat_statements view

Subsequent processutility hooks after bbf_processutilty are not being called.
Created a hook for that and called it inside relevant
ProcessUtility function.

Signed-off-by: Kritika <kritikaq@amazon.com>
